src/server/pfs/server/api_server_new_storage.go

```diff
@@ -10,35 +10,57 @@ import (
 	"github.com/pachyderm/pachyderm/src/client/pfs"
 	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
 	"github.com/pachyderm/pachyderm/src/client/pkg/grpcutil"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/fileset"
 	"github.com/pachyderm/pachyderm/src/server/pkg/storage/metrics"
 )
 
 func (a *apiServer) PutTar(server pfs.API_PutTarServer) (retErr error) {
```

Comment:
It would be nice if this function enforced the protocol a bit more strongly (to help catch misuse earlier).  E.g. error if subsequent requests set the `Commit` field, error if a request has both `EOF` and `Data` set.

Reply:
Sure, but that is something that is not too high priority now, particularly because I am still trying to figure out what will be needed from the API. We are not even in alpha yet lol.